axelor-contract/src/main/java/com/axelor/apps/contract/db/repo/ContractVersionRepository.java

```diff
@@ -17,17 +17,56 @@
  */
 package com.axelor.apps.contract.db.repo;
 
+import com.axelor.apps.contract.db.Contract;
+import com.axelor.apps.contract.db.ContractLine;
 import com.axelor.apps.contract.db.ContractVersion;
-import com.axelor.db.JpaRepository;
+import com.axelor.apps.tool.ModelTool;
+import com.axelor.inject.Beans;
+import java.util.List;
 
-public class ContractVersionRepository extends JpaRepository<ContractVersion> {
+public class ContractVersionRepository extends AbstractContractVersionRepository {
 
-  public static final int DRAFT_VERSION = 1;
-  public static final int WAITING_VERSION = 2;
-  public static final int ONGOING_VERSION = 3;
-  public static final int TERMINATED_VERSION = 4;
+  public ContractVersion copy(Contract contract) {
```

Comment:
Better to use super.copy method. Like that, if we add a specific field inside a specific customer module, the specific field value will be kept.